### PR TITLE
Explain allowable diff in Double.Equals()

### PR DESCRIPTION
**Pull request by gewarren**
Fixes https://devdiv.visualstudio.com/DevDiv/_queries/edit/1925778.

(Hide whitespace changes.)

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/fundamentals/runtime-libraries/system-double-equals.md](https://github.com/dotnet/docs/blob/8057bfdbada4fbf1c8adec0fe2041954a520c65e/docs/fundamentals/runtime-libraries/system-double-equals.md) | [System.Double.Equals method](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-double-equals?branch=pr-en-us-41209) |

### CODE REVIEW

File: docs/fundamentals/runtime-libraries/system-double-equals.md [hunk 1]
```diff
@@ -34,14 +34,21 @@ Rather than comparing for equality, one technique involves defining an acceptabl
...
 
 :::code language="csharp" source="./snippets/System/Double/Equals/csharp/equalsabs1.cs" interactive="try-dotnet" id="Snippet1":::
 :::code language="fsharp" source="./snippets/System/Double/Equals/fsharp/equalsabs1.fs" id="Snippet1":::
 :::code language="vb" source="./snippets/System/Double/Equals/vb/equalsabs1.vb" id="Snippet1":::
 
-The precision of floating-point numbers beyond the documented precision is specific to the implementation and version of the .NET Framework. Consequently, a comparison of two particular numbers might change between versions of the .NET Framework because the precision of the numbers' internal representation might change.
+> [!NOTE]
+> For some values, you might consider them equal even when there *is* a possible floating-point value between the integer representations. For example, consider the double values `0.39` and `1.69 - .3` (which is calculated as `0.3899999999999999`). On a little-endian computer, the integer representations of these values are `4600697235336603894` and `4600697235336603892`, respectively. The difference between the integer values is `2`, meaning there *is* a possible floating-point value between `0.39` and `1.69 - .3`.
```
Comment (gewarren) [inline docs/fundamentals/runtime-libraries/system-double-equals.md:46]: Not sure if there's a more formulaic way to explain this. If so, please comment.

Comment (IEvangelist) [inline docs/fundamentals/runtime-libraries/system-double-equals.md:46]: Verified here within this [sharplab.io](https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUH1TATgAoASAIirwAI6aBJAZxoBsBLYYVgUwFoeUOOwCGUEDQDeAIU4BhAPZQAbjzDA1AOmYAZTtx4BRIaKgBfWvSoUAlAG48eOAogAjXjQAOC9lGAAVAAswHh4AOV8eGgBeGgAGTQBmIgdcZzcPb19gAGV2AA8IqB4AWV8IJgAFHz8gkKjYhIA2Iho+GiTUgmJyCgTkmJoAckks2uDQop4zIdtUwlJKZtb2pMGRsdyCqbKoCursutCZucdcViUAcxp8wdlgRRU1DTBNABEXdx5/BQY/JoALPcmCRNkdwpF7HgLlBrgBPO7yJSqdRaD4Zb6/f5AzggzZ5QqRXb7GoBCY8KH4XALXr3R4ol7vT68H5/YCA4EkfpEGyDST5Mynak9Sh05HPNHMzFsjm4rmaFptDqJXmxSRwwWUoA==).

```suggestion
> For some values, you might consider them equal even when there *is* a possible floating-point value between the `long` representations. For example, consider the `double` values `0.39` and `1.69 - .3` (which is calculated as `0.38999999999999996`). On a little-endian computer, the `long` representations of these values are `4600697235336603894` and `4600697235336603893`, respectively. The difference between the `long` values is `1`, meaning there *is* a possible floating-point value between `0.39` and `1.69 - .3`.
```

**Review by IEvangelist — APPROVED** [review]
Ran this through sharplab.io.

File: docs/fundamentals/runtime-libraries/system-double-equals.md (hunk 1 above)
Comment (gewarren) [reply docs/fundamentals/runtime-libraries/system-double-equals.md:46]: @IEvangelist Those aren't the values I saw on my machine. It probably changes a bit depending on the architecture. Also, if the difference is 1, that means there *isn't* a possible value in between.

Comment (IEvangelist) [reply docs/fundamentals/runtime-libraries/system-double-equals.md:46]: What's the architecture on your machine? Also, regarding the difference of 1, I just assumed a difference at all was what we were calling out in terms of floating-point precision error. In an ideal computing world, these values would match exactly.

Comment (IEvangelist) [reply docs/fundamentals/runtime-libraries/system-double-equals.md:46]: I updated the [sharplab.io](https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUKgBgAJVMA6AJWmAEsBbAUzIEkpgGwB7ABwGUOAbjQDGDAM4BuPHlIBOABQASAESq8RDUWZiiAGxrBguhgFoGUODQCGUEEQDeAIQMBhTlAEd2YFmIAyBkYMAKIW1lAAvuqaqsoAlFK4MpgKKmq4mkQA8rxEVmDCABYGDAAewBBgDHYOVGz0DKwAZpxgdFa07mQ5AIIFxezCFVVRGTGqCdL4KfKTY5kLi0R4cJwQAEbGRNycNGwAKoVVDAByewxEALxEBGQAzLKJqxtbO3vAvDSlZ1AMALJ7CBiAAKuwORwYF2utwAbLIiCYiPdEslUspbg87AByexvcHHH4MCJY+KJORKdFkOEIpF3bG4sEfL6EgFQIGg96HY7E0lTclzJaCzR4XTuADmRFKVyIzmAbg8Xg4ZAAImtNgx9pxWMAYQAWWVieR44BcyGEuaiqASgCe0tl8s8YG8KrVxk12r1BqNjM+33OrPZjNNDDmqIp9vcjudqpeGq1bE9BkNGNkcTs9lKEV50zREYVTqVMfV7oT+qT8lh8MR9zTDmtWbmQA), now it outputs OS arch too. I also replicated this as a [.NET fiddle, with the same results here](https://dotnetfiddle.net/mAboIs).

Comment (gewarren) [inline docs/fundamentals/runtime-libraries/system-double-equals.md:46]:
```suggestion
> For some values, you might consider them equal even when there *is* a possible floating-point value between the integer representations. For example, consider the double values `0.39` and `1.69 - 1.3` (which is calculated as `0.3899999999999999`). On a little-endian computer, the integer representations of these values are `4600697235336603894` and `4600697235336603892`, respectively. The difference between the integer values is `2`, meaning there *is* a possible floating-point value between `0.39` and `1.69 - 1.3`.
```

Comment (gewarren) [reply docs/fundamentals/runtime-libraries/system-double-equals.md]: Oops, it was supposed to be 1.69 - 1.3, not 1.69 - .3. I updated your sharplab with that (I think you had 0.69 - 0.3 before) and now it matches my machine's output.

**Pull request merged**